### PR TITLE
qa/tasks/vstart_runner: use init-ceph to start/stop daemons; do not omit sudo

### PR DESCRIPTION
**Pull request by liewegas**
This allows the mgr/nfs tests to run with vstart_runner. 

- Use init-ceph to start/stop daemons.  The primary advantage here is that all of the paths and environment variables get set up properly by init-ceph, which is well-maintained.  The downside is that if we stop a daemon we can't capture the exit code; I'm not sure if there are any tests that need that.
- Do not omit sudo, as this prevents mount/umount calls from working.  I would rather we be more selective in sprinkling sudo in the test code so that it behaves better in a vstart_runner dev environment... is that feasible?

### CODE REVIEW

**Comment by batrick** [comment]
>     * Do not omit sudo, as this prevents mount/umount calls from working.  I would rather we be more selective in sprinkling sudo in the test code so that it behaves better in a vstart_runner dev environment... is that feasible?

I've tried to effect that for most cephfs tests here: https://github.com/ceph/ceph/pull/42075/commits/e2b39f6c8c23b54d7e1ff326175b371a7496f85e

File: qa/tasks/vstart_runner.py [hunk 1]
```diff
@@ -416,6 +416,7 @@ def run(self, **kwargs):
     def _do_run(self, args, check_status=True, wait=True, stdout=None,
                 stderr=None, cwd=None, stdin=None, logger=None, label=None,
                 env=None, timeout=None, omit_sudo=True, shell=True):
+        omit_sudo = False
```
Comment (batrick) [inline qa/tasks/vstart_runner.py:419]: This is to make `qa/tasks/cephfs/test_nfs.py` work, right? I think the patch required is:

````
diff --git a/qa/tasks/cephfs/test_nfs.py b/qa/tasks/cephfs/test_nfs.py
index 3013dffe1e5..e42c3dab0d1 100644
--- a/qa/tasks/cephfs/test_nfs.py
+++ b/qa/tasks/cephfs/test_nfs.py
@@ -256,7 +256,7 @@ class TestNFS(MgrTestCase):
         '''
         try:
             self.ctx.cluster.run(args=['sudo', 'mount', '-t', 'nfs', '-o', f'port={port}',
-                                       f'{ip}:{pseudo_path}', '/mnt'])
+                                       f'{ip}:{pseudo_path}', '/mnt'], omit_sudo=False)
         except CommandFailedError as e:
             # Check if mount failed only when non existing pseudo path is passed
             if not check and e.exitstatus == 32:
````

and elsewhere.

**Review by batrick — CHANGES_REQUESTED** [review]

**Review by batrick — COMMENTED** [review]
Changes look good to me but you have some jenkins test failures.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-07-17_04:37:01-rados-wip-kefu-testing-2021-07-17-1006-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-07-17_15:53:41-rados-wip-kefu-testing-2021-07-17-1518-distro-basic-smithi/

one of the failures is tracked by:

- https://tracker.ceph.com/issues/50245

the e2e failures are not related.

**Comment by tchaikov** [comment]
jenkins test api

**Comment by tchaikov** [comment]
jenkins test api

**Comment by tchaikov** [comment]
```
FAIL: test_host_daemons (tasks.mgr.dashboard.test_host.HostControllerTest)
...
2021-07-18 03:50:10,239.239 INFO:__main__:----------------------------------------------------------------------
2021-07-18 03:50:10,239.239 INFO:__main__:Traceback (most recent call last):
2021-07-18 03:50:10,240.240 INFO:__main__:  File "/home/jenkins-build/build/workspace/ceph-api/qa/tasks/mgr/dashboard/test_host.py", line 95, in test_host_daemons
2021-07-18 03:50:10,240.240 INFO:__main__:    self.assertStatus(200)
2021-07-18 03:50:10,240.240 INFO:__main__:  File "/home/jenkins-build/build/workspace/ceph-api/qa/tasks/mgr/dashboard/helper.py", line 494, in assertStatus
2021-07-18 03:50:10,240.240 INFO:__main__:    self.assertEqual(self._resp.status_code, status)
2021-07-18 03:50:10,240.240 INFO:__main__:AssertionError: 500 != 200
```

**Review by epuertat — APPROVED** [review]
Looks good! I see you modified the test_orchestrator. Is it still maintained (@sebastian-philipp)? At the Dashboard team, besides the Docker-in-docker cephadm env, we were thinking on revamping some fake orchestrator implementation for quick vstart testing.

BTW I resolved the conflicting rebase in the GH website, and it created a merge PR... I guess that will disappear when merging back to master, but I'm not familiar with this process :/

**Pull request closed without merging**